tools/clang/tools/dxcompiler/dxcpdbutils.cpp

```diff
@@ -65,21 +70,143 @@ static HRESULT CopyWstringToBSTR(const std::wstring &str, BSTR *pResult) {
   return S_OK;
 }
 
-static bool ShouldIncludeInFlags(StringRef strRef, bool *skip_next_arg) {
+static std::wstring NormalizePath(const WCHAR *path) {
+  std::string FilenameStr8 = Unicode::UTF16ToUTF8StringOrThrow(path);
+  llvm::SmallString<128> NormalizedPath;
+  llvm::sys::path::native(FilenameStr8, NormalizedPath);
+  std::wstring FilenameStr16 = Unicode::UTF8ToUTF16StringOrThrow(NormalizedPath.c_str());
+  return FilenameStr16;
+}
+
+static bool IsBitcode(const void *ptr, size_t size) {
+  const uint8_t pattern[] = {'B','C',};
+  if (size < _countof(pattern))
+    return false;
+  return !memcmp(ptr, pattern, _countof(pattern));
+}
+
+static bool ShouldIncludeInFlags(const std::wstring &str, bool *skip_next_arg) {
   *skip_next_arg = false;
-  const char *specialCases[] = { "/T", "-T", "-D", "/D", "-E", "/E", };
+  const wchar_t *specialCases[] = { L"/T", L"-T", L"-D", L"/D", L"-E", L"/E", };
```

Comment:
Of course this should be all options that take separate arguments.  Or ideally, use llvm::opt::OptTable::ParseArgs to parse them correctly instead.

Reply:
Oh, right, this is for stripping these options from the other ones.  Still, same ParseArgs can be used, followed by render() for each arg that isn't one of these into ArgStringList.  Of course this is merely to satisfy the legacy interface that expresses these options separately from the option list, which PIX currently expects.  Better would be to make a canonical representation of all arguments that can be processed without special knowledge about each argument.

Reply:
This was removed in favour of parsing everything properly with opt table (for the old debug module case)